mypy/util.py

```diff
@@ -906,3 +919,20 @@ def quote_docstring(docstr: str) -> str:
         return f"''{docstr_repr}''"
     else:
         return f'""{docstr_repr}""'
+
+
+_default = object()
```

Comment:
if you want to over-engineer this, then the closest you could get to a typing-compatible sentinel is with something like

```python
@final
class _DefaultEnum(enum.Enum):
    default = object()

DefaultType: TypeAlias = Literal[_DefaultEnum.default]
Default: Final[DefaultType] = _DefaultEnum.default
```

so that you can do stuff like

```python
def maybe(obj: T | DefaultType = Default, /) -> T | None: ...
```

Reply:
`Enum`s are already `final`

I would love if this sentinel was understood by typecheckers

Reply:
> `Enum`s are already `final`

I didn't know that 😮 !

Reply:
only if they define members

Reply:
is an enum without members even an enum 🤔 ?

Reply:
no, that's why it's not `final` :)